### PR TITLE
Allow partially defined dimensions to read/serialize from/to IR

### PR DESCRIPTION
**Pull request by lazarevevgeny**
This PR enables reading/serializing IRs with -1 in dimensions. However we still keep the transformation which makes partially defined shapes static when upper bound is known (for example, produced with NMS operation). This is done in order to not break POT functionality which relies on the MO which is not able to read IRs with `-1`s. Once the PR #5918 is merged these restrictions can be removed.

This is needed for so many new features so I have decided to not include the list of tickets :)

### CODE REVIEW

File: inference-engine/tests/functional/inference_engine/ir_serialization/cleanup.cpp [hunk 1]
```diff
@@ -49,15 +49,15 @@ TEST_F(SerializationCleanupTest, SerializationShouldWork) {
     ASSERT_TRUE(std::ifstream(m_out_bin_path, std::ios::in).good());
 }
 
-TEST_F(SerializationCleanupTest, SerializationShouldFail) {
+TEST_F(SerializationCleanupTest, SerializationShouldWorkWithDynamicFunction) {
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/ir_serialization/cleanup.cpp:52]: can we add explicit test that reads IR with PDS?

Comment (lazarevevgeny) [reply inference-engine/tests/functional/inference_engine/ir_serialization/cleanup.cpp:52]: This is exactly such a test! :) 
We create a function with an partially defined dimension in line 55.

Comment (lazarevevgeny) [reply inference-engine/tests/functional/inference_engine/ir_serialization/cleanup.cpp:52]: Oh. Sorry, I see what you mean.

File: inference-engine/tests/functional/inference_engine/ir_serialization/models/dynamic_input_shape.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0"?>
+<net name="Function_0" version="10">
+	<layers>
+		<layer id="0" name="Parameter_68" type="Parameter" version="opset1">
+			<data cacheable="false" shape="1,-1,3" element_type="f32" />
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/ir_serialization/models/dynamic_input_shape.xml:5]: do we use `cacheable`?

Comment (lazarevevgeny) [reply inference-engine/tests/functional/inference_engine/ir_serialization/models/dynamic_input_shape.xml]: Copy-paste issue. Fixed.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by lazarevevgeny** [comment]
This PR cannot be merged because POT will fail to load models with `-1` in the dimension. This is the case for models with NMS operation which has dynamic output shape after model serialization with the IE at the MO back phase.

**Pull request merged**